### PR TITLE
Major dependency upgradation - inprogress

### PR DESCRIPTION
**Pull request by anasnadeemws**
## Summary by CodeRabbit

- **Chores**
	- Upgraded GitHub Actions workflow actions for enhanced performance and security.
	- Updated testing libraries to their latest versions for improved compatibility and functionality.
	- Streamlined import statements in the testing setup for better clarity and organization.
- **Configuration Changes**
	- Adjusted asset prefix handling in the configuration to improve asset resolution.
	- Introduced a comprehensive ESLint configuration to enhance code quality and maintainability.
	- Added PropTypes validation to key components for improved robustness and maintainability.
	- Refactored URL parsing logic in the server configuration for modern best practices.
	- Removed Ant Design CSS import in favor of PropTypes for the MyApp component.

### CODE REVIEW

File: app/utils/tests/injectSaga.test.js [hunk 1]
```diff
@@ -1,131 +1,137 @@
-/* eslint-disable import/namespace */
 /* eslint-disable no-import-assign */
+/* eslint-disable react/display-name */
```
Comment (praveenkumar1798) [inline app/utils/tests/injectSaga.test.js:2]: Why are we disabling eslint rules ?

File: eslint.config.mjs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  recommendedConfig: js.configs.recommended,
+  allConfig: js.configs.all
+});
+
+const prettierOptions = JSON.parse(fs.readFileSync(path.resolve(__dirname, '.prettierrc'), 'utf8'));
+
+export default [
+  ...compat.extends(
+    'eslint:recommended',
+    // 'plugin:import/recommended',
```
Comment (praveenkumar1798) [inline eslint.config.mjs]: Why are these commented ?

File: eslint.config.mjs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    'plugin:react/recommended',
+    'plugin:jest/recommended',
+    'plugin:prettier/recommended',
+    // 'next',
```
Comment (praveenkumar1798) [inline eslint.config.mjs]: Why is this commented ?

File: package.json [hunk 4]
```diff
@@ -40,58 +40,84 @@
...
+    "prettier": "^3.3.3",
+    "prettier-config-standard": "^7.0.0",
     "react-floki": "^1.0.96",
     "react-test-renderer": "^18.3.1",
-    "stylelint": "^14.16.1"
+    "stylelint": "^16.9.0",
+    "stylelint-config-recommended": "^14.0.1",
+    "stylelint-config-standard-scss": "^13.1.0",
+    "stylelint-config-styled-components": "^0.1.1",
+    "typescript": "^5.5.4"
```
Comment (praveenkumar1798) [inline package.json]: Are we using typescript here ?

File: package.json [hunk 5]
```diff
@@ -3,9 +3,9 @@
   "version": "0.1.0",
   "private": true,
   "scripts": {
-    "build:prod": "env-cmd -f environments/.env.production next build && next export",
-    "build:qa": "env-cmd -f environments/.env.qa next build && next export",
-    "build:dev": "env-cmd -f environments/.env.development next build && next export",
```
Comment (praveenkumar1798) [inline package.json:8]: Did you check if you were able to serve your build and if it works as expected. next export will export all your pages to static HTML files that you can serve with any host. So please check if  it is okay to remove

**Review by praveenkumar1798 — CHANGES_REQUESTED** [review]
The CI is failing because of the lint. Please fix it along with other comments added

File: package.json (hunk 5 above)
Comment (anasnadeemws) [reply package.json:8]: Screenshot 2024-09-16 at 11 43 20 AM
`next export` has been removed in favor of 'output: export' in next.config.js

Yes I was able to build and server, please check below ss
Screenshot 2024-09-16 at 11 45 20 AM

File: app/utils/tests/injectSaga.test.js (hunk 1 above)
Comment (anasnadeemws) [reply app/utils/tests/injectSaga.test.js:2]: import/no-name-as-default doesn’t work with eslint 9
https://github.com/import-js/eslint-plugin-import/issues/2995

So for now we need to comment it, till the community resolves it

**Pull request merged**